cli/cmd/top.go

```diff
@@ -0,0 +1,389 @@
+		"Display requests with paths that start with this prefix")
+
+	return cmd
+}
+
+func getTrafficByResourceFromAPI(w io.Writer, client pb.ApiClient, req *pb.TapByResourceRequest) error {
+	rsp, err := client.TapByResource(context.Background(), req)
+	if err != nil {
+		return err
+	}
+
+	err = termbox.Init()
+	if err != nil {
+		return err
+	}
+	defer termbox.Close()
+
+	requestCh := make(chan topRequest, 100)
+	done := make(chan struct{})
+
+	go recvEvents(rsp, requestCh, done)
+	go pollInput(done)
+
+	renderTable(requestCh, done)
+
+	return nil
+}
+
+func recvEvents(tapClient pb.Api_TapByResourceClient, requestCh chan<- topRequest, done chan<- struct{}) {
+	outstandingRequests := make(map[topRequestID]topRequest)
+	for {
+		event, err := tapClient.Recv()
+		if err == io.EOF {
+			fmt.Println("Tap stream terminated")
+			close(done)
+			return
+		}
+		if err != nil {
+			fmt.Println(err.Error())
+			close(done)
+			return
+		}
+		id := topRequestID{
+			src: addr.PublicAddressToString(event.GetSource()),
+			dst: addr.PublicAddressToString(event.GetDestination()),
+		}
+		switch ev := event.GetHttp().GetEvent().(type) {
+		case *pb.TapEvent_Http_RequestInit_:
+			id.stream = ev.RequestInit.GetId().Stream
+			outstandingRequests[id] = topRequest{
+				event:   event,
+				reqInit: ev.RequestInit,
+			}
+
+		case *pb.TapEvent_Http_ResponseInit_:
+			id.stream = ev.ResponseInit.GetId().Stream
+			if req, ok := outstandingRequests[id]; ok {
+				req.rspInit = ev.ResponseInit
+				outstandingRequests[id] = req
+			} else {
+				log.Warnf("Got ResponseInit for unknown stream: %s", id)
+			}
+
+		case *pb.TapEvent_Http_ResponseEnd_:
+			id := topRequestID{
+				addr.PublicAddressToString(event.GetSource()),
+				addr.PublicAddressToString(event.GetDestination()),
+				ev.ResponseEnd.GetId().Stream,
+			}
+			if req, ok := outstandingRequests[id]; ok {
+				req.rspEnd = ev.ResponseEnd
+				requestCh <- req
+			} else {
+				log.Warnf("Got ResponseEnd for unknown stream: %s", id)
+			}
+		}
+	}
+}
+
+func pollInput(done chan<- struct{}) {
+	for {
+		switch ev := termbox.PollEvent(); ev.Type {
+		case termbox.EventKey:
+			if ev.Ch == 'q' || ev.Key == termbox.KeyCtrlC {
+				close(done)
+				return
+			}
+		}
+	}
+}
+
+func renderTable(requestCh <-chan topRequest, done <-chan struct{}) {
+	ticker := time.NewTicker(100 * time.Millisecond)
+	var table []tableRow
+
+	for {
+		select {
+		case <-done:
+			return
+		case req := <-requestCh:
+			tableInsert(&table, req)
+		case <-ticker.C:
+			termbox.Clear(termbox.ColorDefault, termbox.ColorDefault)
+			renderHeaders()
+			renderTableBody(&table)
+			termbox.Flush()
+		}
+	}
+}
+
+func tableInsert(table *[]tableRow, req topRequest) {
+	by := req.reqInit.GetPath()
+	source := stripPort(addr.PublicAddressToString(req.event.GetSource()))
+	destination := stripPort(addr.PublicAddressToString(req.event.GetDestination()))
+	if pod := req.event.DestinationMeta.Labels["pod"]; pod != "" {
+		destination = pod
+	}
```

Comment:
Per-in person conversation, I think we should also display the source pod if it's available in SourceMeta, similar to what you're already doing for the destination pod.